realm/realm-annotations-processor/src/main/java/io/realm/processor/ClassMetaData.java

```diff
@@ -332,11 +337,13 @@ private boolean checkDefaultConstructor() {
 
     private boolean checkForFinalFields() {
         for (VariableElement field : fields) {
-            if (field.getModifiers().contains(Modifier.FINAL)) {
-                Utils.error(String.format(Locale.US,
-                        "Class \"%s\" contains illegal final field \"%s\".", className, field.getSimpleName().toString()));
-                return false;
-            }
+            if (!field.getModifiers().contains(Modifier.FINAL)) { continue; }
```

Comment:
We introduced an automatic formatter that allows this kind of code. No-one on the team found it hindered readability, nor could lead to errors.